src/Sentry/Internal/MainSentryEventProcessor.cs

```diff
@@ -133,6 +135,36 @@ public SentryEvent Process(SentryEvent @event)
             return @event;
         }
 
+        private void AddMemoryInfo(Contexts contexts)
+        {
+#if NETCOREAPP3_0_OR_GREATER
+            var memoryInfo = GC.GetGCMemoryInfo();
+            var values = new Dictionary<string, string>
+            {
+                {"TotalAllocatedBytes", GC.GetTotalAllocatedBytes().ToString()},
+                {"FragmentedBytes", memoryInfo.FragmentedBytes.ToString()},
+                {"HeapSizeBytes", memoryInfo.HeapSizeBytes.ToString()},
+                {"HighMemoryLoadThresholdBytes", memoryInfo.HighMemoryLoadThresholdBytes.ToString()},
+                {"TotalAvailableMemoryBytes", memoryInfo.TotalAvailableMemoryBytes.ToString()},
+                {"MemoryLoadBytes", memoryInfo.MemoryLoadBytes.ToString()},
+            };
+#if NET5_0_OR_GREATER
+            values.Add("TotalCommittedBytes", memoryInfo.TotalCommittedBytes.ToString());
+            values.Add("PromotedBytes", memoryInfo.PromotedBytes.ToString());
+            values.Add("PinnedObjectsCount", memoryInfo.PinnedObjectsCount.ToString());
+            values.Add("PauseTimePercentage", memoryInfo.PauseTimePercentage.ToString(CultureInfo.InvariantCulture));
+            var pauseDurations = memoryInfo.PauseDurations.ToArray();
+            values.Add("PauseDurations", string.Join(';', pauseDurations.Select(x => x.ToString())));
```

Comment:
nitpick: prefer the first character of the 'thing' the collection contains.

```suggestion
            values.Add("PauseDurations", string.Join(';', pauseDurations.Select(p => p.ToString())));
```